wkcuber/tile_cubing.py

```diff
@@ -1,60 +1,204 @@
+    return pattern
 
-    all_source_files.sort()
-    return all_source_files
 
+def get_digit_numbers_for_dimension(pattern):
+    x_number = 0
+    y_number = 0
+    z_number = 0
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    for occurrence in occurrences:
+        if occurrence[1] == "x":
+            x_number = max(x_number, len(occurrence) - 2)
+        if occurrence[1] == "y":
+            y_number = max(y_number, len(occurrence) - 2)
+        if occurrence[1] == "z":
+            z_number = max(z_number, len(occurrence) - 2)
+    return x_number, y_number, z_number
+
+
+def detect_interval_for_dimensions(
+    file_path_pattern: str,
+    x_decimal_length: int,
+    y_decimal_length: int,
+    z_decimal_length: int,
+) -> Tuple[int, int, int, int, int, int, str, int]:
+    arbitrary_file = None
+    file_count = 0
+    decimal_length = {
+        "x": x_decimal_length,
+        "y": y_decimal_length,
+        "z": z_decimal_length,
+    }
+    current_decimal_length = {"x": 0, "y": 0, "z": 0}
+    max_dimensions = {"x": 0, "y": 0, "z": 0}
+    min_dimensions = {"x": None, "y": None, "z": None}
+
+    # find all files by trying all combinations of dimension lengths
+    for x in range(x_decimal_length + 1):
+        current_decimal_length["x"] = x
+        for y in range(y_decimal_length + 1):
+            current_decimal_length["y"] = y
+            for z in range(z_decimal_length + 1):
+                current_decimal_length["z"] = z
+                specific_pattern = replace_coordinates_with_glob_regex(
+                    file_path_pattern, {"z": z, "y": y, "x": x}
+                )
+                found_files = glob(specific_pattern)
+                file_count += len(found_files)
+                for file in found_files:
+                    occurrences = re.findall("({x+}|{y+}|{z+})", file_path_pattern)
+                    index_offset_caused_by_brackets_and_specific_length = 0
+                    for occurrence in occurrences:
+                        # update the offset since the pattern and the file path have different length
+                        occurrence_begin_index = (
+                            file_path_pattern.index(occurrence)
+                            - index_offset_caused_by_brackets_and_specific_length
+                        )
+                        index_offset_caused_by_brackets_and_specific_length += 2
+                        current_dimension = occurrence[1]
+                        occurrence_end_index = (
+                            occurrence_begin_index
+                            + decimal_length[current_dimension]
+                            - current_decimal_length[current_dimension]
+                        )
+                        index_offset_caused_by_brackets_and_specific_length += current_decimal_length[
+                            current_dimension
+                        ]
+                        coordinate_value = int(
```

Comment:
The above code seems quite complicated to me. If I understand correctly, you want to extract the actual values for x, y and z and you already know how many digits each component has. Maybe the following approach is shorter and/or easier to understand?
```python
# Turn a pattern {xxx}/{yyy}/{zzzzzz} for given dimension counts into (e.g., 2, 2, 3) into
# something like xx/yy/zzz (note that the curly braces are gone)
applied_fpp = replace_coordinates(file_path_pattern, {"x": x, "y": y, "z": z})

# For each dimension, look up where it starts within the applied pattern.
# Use that index to look up the actual value within the file name
for current_dimension in ["x", "y", "z"]:
	idx = applied_fpp.index(current_dimension)
	value = int(file_name[idx:idx + decimal_length[current_dimension]])
	assert value not None
	min_dimensions[current_dimension] = min(...)
	max_dimensions[current_dimension] = min(...)
```

(I didn't test the above code, but I hope the gist is correct :))